account_invoice_production_lot/models/account_invoice.py

```diff
@@ -31,10 +31,18 @@ class AccountInvoiceLine(models.Model):
         compute='_compute_line_lots',
     )
 
+    @api.multi
+    def _get_location_id(self):
+        self.ensure_one()
+        order = self.sale_line_ids[0].order_id
+        return order.partner_shipping_id.property_stock_customer
+
     @api.multi
     def _compute_prod_lots(self):
         for line in self:
-            line.prod_lot_ids = line.move_line_ids.mapped('lot_ids')
+            quant_ids = line.move_line_ids.mapped('quant_ids').filtered(
+                lambda x: x.location_id == line._get_location_id())
```

Comment:
Get the location once, outside the `filtered` method: otherwise here it might get uselessly called many times (once for each quant) for each line.

Reply:
Done. See: https://github.com/OCA/account-invoice-reporting/compare/808b4d36e3264296f1a08589c7d22e9d2cc6e5ca..f37f03e8af1380685e31ccc7b91d6d7d609fe259